tests/Task/TaskMetaDataTest.php

```diff
@@ -16,18 +17,18 @@ public static function provideValidUniqueTaskIds () : iterable
 		yield "plain" => ["test"];
 		yield "with dash" => ["a-b"];
 		yield "with underscore" => ["a-b_c"];
-		yield "all characters" => ["a-b_c.d"];
+		yield "all characters" => ["a-b_c.d", "a-b_c-d"];
 		yield "numbers" => ["5"];
 		yield "numbers longer" => ["1-2-3-4"];
 	}
 
 	/**
-	 * @dataProvider provideValidUniqueTaskIds
 	 */
-	public function testValidUniqueTaskIds (string $uniqueTaskId) : void
+	#[DataProvider("provideValidUniqueTaskIds")]
+	public function testValidUniqueTaskIds (string $uniqueTaskId, ?string $key = null) : void
 	{
 		$metadata = new TaskMetaData("Test", uniqueTaskId: $uniqueTaskId);
-		self::assertSame($uniqueTaskId, $metadata->getKey());
+		self::assertSame($key ?? $uniqueTaskId, $metadata->getKey());
```

Comment:
This should be changed to always map the task id to the expected outcome.
The parameter should then be named `$expected`.